ofrak_core/ofrak/gui/server.py

```diff
@@ -260,6 +263,44 @@ async def run_until_cancelled(self):  # pragma: no cover
         finally:
             await self.runner.cleanup()
 
+    @exceptions_to_http(SerializedError)
+    async def root_resource_chunk(self, request: Request) -> Response:
+        name = request.query.get("name")
+        addr = request.query.get("addr")
+        if name is None:
+            raise HTTPBadRequest(reason="Missing resource name from request")
+        if addr is None:
+            raise HTTPBadRequest(reason="Missing chunk address from request")
+        if name not in self.resource_builder.keys():
+            self.resource_builder[name] = {}
+        chunk_data = await request.read()
+        self.resource_builder[name][int(addr)] = chunk_data
```

Comment:
Doing this dictionary keying based on filename seems risky and prone to collisions... 

Can you create a resource first, get its ID, and then update its data after everything has been uploaded?